geoengine/workflow_builder/operators.py

```diff
@@ -855,25 +855,102 @@ def from_operator_dict(cls, operator_dict: Dict[str, Any]) -> 'TimeShift':
         )
 
 
+class RenameBands:
+    '''Base class for renaming bands of a raster.'''
+
+    @abstractmethod
+    def to_dict(self) -> Dict[str, Any]:
+        pass
+
+    @classmethod
+    def from_dict(cls, rename_dict: Dict[str, Any]) -> 'RenameBands':
+        '''Returns a RenameBands object from a dictionary.'''
+        if rename_dict["type"] == "default":
+            return RenameBandsDefault()
+        if rename_dict["type"] == "suffix":
+            return RenameBandsSuffix(cast(List[str], rename_dict["values"]))
+        if rename_dict["type"] == "rename":
+            return RenameBandsRename(cast(List[str], rename_dict["values"]))
+        raise ValueError("Invalid rename type")
+
+    @classmethod
+    def default(cls) -> 'RenameBands':
+        return RenameBandsDefault()
+
+    @classmethod
+    def suffix(cls, values: List[str]) -> 'RenameBands':
+        return RenameBandsSuffix(values)
+
+    @classmethod
+    def rename(cls, values: List[str]) -> 'RenameBands':
+        return RenameBandsRename(values)
+
+
+class RenameBandsDefault(RenameBands):
+    '''Rename bands with default suffix.'''
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {
+            "type": "default"
+        }
+
+
+class RenameBandsSuffix(RenameBands):
+    '''Rename bands with custom suffixes.'''
+
+    suffixes: List[str]
+
+    def __init__(self, suffixes: List[str]) -> None:
+        self.suffixes = suffixes
+        super().__init__()
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {
+            "type": "suffix",
+            "values": self.suffixes
+        }
+
+
+class RenameBandsRename(RenameBands):
+    '''Rename bands with new names.'''
+
+    new_names: List[str]
+
+    def __init__(self, new_names: List[str]) -> None:
+        self.new_names = new_names
+        super().__init__()
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {
+            "type": "suffix",
+            "values": self.new_names
```

Comment:
the type should not be suffix here

Reply:
6458cd32c4c7106d7749c022ede354bf9a883728